pdg/src/graph.rs

```diff
@@ -15,47 +15,85 @@ use crate::util::ShortOption;
 
 #[derive(Debug, Eq, PartialEq, Hash, Clone)]
 pub enum NodeKind {
-    /// A copy from one local to another.  This also covers casts such as `&mut T` to `&T` or `&T`
-    /// to `*const T` that don't change the type or value of the pointer.
+    /// A [`Copy`](Self::Copy) from one [`Local`] to another.
+    ///
+    /// This also covers casts such as `&mut T` to `&T` or `&T` to `*const T`
+    /// that don't change the type or value of the pointer.
     Copy,
 
-    /// Field projection.  Used for operations like `_2 = &(*_1).0`.  Nested field accesses like
-    /// `_4 = &(*_1).x.y.z` are broken into multiple `Node`s, each covering one level.
+    /// [`Field`] projection.
+    ///
+    /// Used for operations like `_2 = &(*_1).0`.
+    /// Nested field accesses like `_4 = &(*_1).x.y.z`
+    /// are broken into multiple [`Node`]s, each covering one level.
     Field(Field),
-    /// Pointer arithmetic.  The `isize` is the concrete offset distance.  We use this to detect
-    /// when two pointers always refer to different indices.
+
+    /// Pointer arithmetic.
+    ///
+    /// The [`isize`] is the concrete offset distance.
+    /// We use this to detect when two pointers always refer to different indices.
     Offset(isize),
 
     // Operations that can't have a `source`.
```

Comment:
What's this referring to?

Reply:
it refers to `Node::source`, which will be `None` for all `Node`s with any of the following `NodeKind`s

Reply:
I meant what is the comment and operation referring to.  Which variant?

Reply:
all of them that follow until the next similar comment

Reply:
Ahh, got it.  I'll make that clearer.

Reply:
I'll put that in the docs for each variant it applies to, as I think doing it as is is a bit brittle if the variants are ever moved around, and it wasn't that clear to begin with.

Reply:
Fixed in https://github.com/immunant/c2rust/pull/658/commits/dc29cfaabeb0dbc8764247e84decdad82ad3f3e8.